core/src/main/java/com/netflix/conductor/service/ExecutionService.java

```diff
@@ -123,10 +123,18 @@ public Task poll(String taskType, String workerId, String domain) {
 		}
 		String queueName = QueueUtils.getQueueName(taskType, domain, null,null);
 
+		List<String> taskIds = new LinkedList<>();
 		List<Task> tasks = new LinkedList<>();
 		try {
-			List<String> taskIds = queueDAO.pop(queueName, count, timeoutInMilliSecond);
-			for (String taskId : taskIds) {
+			taskIds = queueDAO.pop(queueName, count, timeoutInMilliSecond);
+		} catch (Exception e) {
+			logger.error("Error polling for task: {} from worker: {} in domain: {}, count: {}", taskType, workerId,
+				domain, count, e);
+			Monitors.error(this.getClass().getCanonicalName(), "taskPoll");
```

Comment:
How about surfacing this exception to client? Also, should we update `updateTaskLastPoll` if the pop fails?

Reply:
Yes, this helps to track polling from workers irrespective of whether a task was returned in the poll.

Reply:
This exception was not surfaced to the client in the previous implementation either. Since polling is a highly-frequent activity, it would clutter up client-side logs in case of a temporary dependency glitch on server-side.